### DISCUSSION TITLE
Search for X

### DISCUSSION THREAD

**Discussion by xxxserxxx** [spezifisch/stmps#28] [Ideas]
This relates to #26, which is a more broad search function.

I'm opening this discussion because, while the search itself probably won't be challenging, how the results are presented to the user should be generally agreed advance.

I'm thinking a new Search page with multiple columns for the common top-level attributes: Artist, Album, Genre, and (song) Title. I'd like the search itself to be general: 

1. the user types anything into the search field and hits enter
2. stmps queries the server and gets all results
3. stmps filters the results into the columns based on how they matched
4. the user can navigate the columns and add things to the Queue as they would in the browser

**Comment by spezifisch** [comment]
Regarding usability: I forget all the time that the search input field automatically focuses when switching to the search page.

What I'm trying to do:
- Key '4': play something from the search page
- Key '2': look at queue
- Key '4': i want to add another album from the selected artist, but the search field is focused
- Arrow Keys: Trying to select the next album, not realizing I'm not in the middle column where I was.
- ESC
- now arrow keys work, of course

Am I using this wrong?

**Comment by xxxserxxx** [comment]
Bad design decision on my part. It breaks with the common behavior of other tabs. I agree that it should not be selected by default and, in fact, maybe require the same "search" hotkey to activate as in the browser tab.

It's currently bad, and should mirror search in `page_browser`, even if the search field is always visible (rather than popping in and out as in `page_browser`). It's a simple change, I think: we just change the default focused widget to the artist column -- or slightly less simple, the first column that contains any elements.

I'm going to fix the image loading performance thing first; if you don't get to this before I do, I'll make the change. It should affect only a couple of lines of code.

**Comment by xxxserxxx** [comment]
This is resolved in one of the branches -- both the search for X feature, and the irregular default widget focus.

With the search page, we can search for artists, albums, or songs in one go. With the genre search (which, oddly, `/search3` does _not_ search for), users can browse and search genres. I think this covers 90% of things users would like to search for. I suppose we could consider other metadata: year, or I don't know what else, but I feel as if that's getting a bit feature-creep-ish.

Consequently, I'm going to close this conversation. The actual feature tickets must wait until the PRs are merged, but I don't think there's more to discuss.
